lib/chef/provisioning/aws_driver/super_lwrp.rb

```diff
@@ -7,30 +7,43 @@ class SuperLWRP < Chef::Resource::LWRPBase
     #
     # Add the :lazy_default and :coerce validation_opts to `attribute`
     #
-    def self.attribute(attr_name, validation_opts={})
-      lazy_default = validation_opts.delete(:lazy_default)
-      coerce = validation_opts.delete(:coerce)
-      if lazy_default || coerce
-        define_method(attr_name) do |arg=nil|
-          arg = instance_exec(arg, &coerce) if coerce && !arg.nil?
+    if self.respond_to?(:properties)
+      # in Chef 12.4+, properties replace attributes and these respond to
+      # coerce and default with a lazy block
+      def self.attribute(attr_name, validation_opts={})
+        lazy_default = validation_opts.delete(:lazy_default)
+        if lazy_default
+          Chef.log_deprecation("#{name} defines a lazy_default validation, instead specify default with a lazy block when Chef supports properties")
```

Comment:
In Provisioning, this is the syntax we use for lazy defaults:

```
attribute :x, lazy_default: proc { ... }
```

In 12.5 this is the syntax you use:

```
attribute :x, default: lazy { ... }
```

In the 12.5 side of that `if` statement, you are adding support to convert `lazy_default` to `default` inside `SuperLWRP`. I'm suggesting that instead, you remove `def self.attribute` entirely in 12.5 and convert all callers in `chef-provisioning-aws` to 12.5 syntax (`default: lazy { ... }`). This would just mean that the 12.4 side would have to stop directly supporting `lazy_default` as an option and instead recognize `default: lazy { ... }` and support that directly.